src/ServicePulse.Host/vue/src/components/failedmessages/FlowDiagram.vue

```diff
@@ -61,292 +58,176 @@ function mapMessage(message) {
+      .reduce((groups, message) => {
+        groups[message.level] = [...(groups[message.level] ?? []), message];
+        return groups;
+      }, [])
+      //ensure each level has their items in the same "grouped" order as the level above
+      .map((group, level, messagesByLevel) => {
+        const previousLevel = level > 0 ? messagesByLevel[level - 1] : null;
+        return group.sort(
+          (a, b) =>
+            (previousLevel?.findIndex((plMessage) => a.parentId === plMessage.messageId && a.parentEndpoint === plMessage.receivingEndpoint) ?? 1) -
+            (previousLevel?.findIndex((plMessage) => b.parentId === plMessage.messageId && b.parentEndpoint === plMessage.receivingEndpoint) ?? 1)
+        );
+      })
+      //flatten to actual flow diagram nodes, with positioning based on parent node/level
+      .flatMap((group, level, messagesByLevel) => {
+        const previousLevel = level > 0 ? messagesByLevel[level - 1] : null;
+        return group.reduce(
+          ({ result, currentWidth, previousParent }, message) => {
+            //position on current level needs to be based on parent Node, so see if one exists
+            const parentMessage = previousLevel?.find((plMessage) => message.parentId === plMessage.messageId && message.parentEndpoint === plMessage.receivingEndpoint);
+            //if the current parent node is the same as the previous parent node, then the current position needs to be to the right of siblings
+            const currentParentWidth = previousParent === parentMessage ? currentWidth : 0;
+            const startX = parentMessage == null ? 0 : parentMessage.XPos - parentMessage.width / 2;
+            //store the position of the node against the message, so child nodes can use it to determine their start position
+            message.XPos = startX + (currentParentWidth + message.width / 2);
+            return {
+              result: [
+                ...result,
+                {
+                  id: `${message.messageId}##${message.receivingEndpoint}`,
+                  type: "message",
+                  data: message,
+                  label: message.nodeName,
+                  position: { x: message.XPos * 300, y: message.level * 200 },
+                },
+              ],
+              currentWidth: currentParentWidth + message.width,
+              previousParent: parentMessage,
+            };
+          },
+          { result: [], currentWidth: 0, previousParent: null }
+        ).result;
+      })
+  );
+}
+
+function constructEdges(mappedMessages) {
+  return mappedMessages
+    .filter((message) => message.parentId)
+    .map((message) => ({
+      id: `${message.parentId}##${message.messageId}`,
+      source: `${message.parentId}##${message.parentEndpoint}`,
+      target: `${message.messageId}##${message.receivingEndpoint}`,
+      markerEnd: MarkerType.ArrowClosed,
+      style: {
+        "stroke-dasharray": message.type === "Event message" && "5, 3",
```

Comment:
So should this have a `[]` for the else condition?

Reply:
Also where does 5, 3 come from?

Reply:
> So should this have a `[]` for the else condition?

I don't follow?

> Also where does 5, 3 come from?

The old style for links (which never worked)

Reply:
I think this is an idiomatic shorthand for

> If the message type is "Event message" then set `stroke-dasharray` to "5, 3". Otherwise do not set it.

Is that right? I noticed the pattern in a couple of places and it took me a second to parse it as well.

Reply:
yes

Reply:
Well usually 
```
message.type === "Event message" && "5, 3",
```
results in either `"5, 3"` or `false`
So is the intent here to set `"stroke-dasharray": false` ?

Reply:
Here is what I mean - https://www.typescriptlang.org/play?#code/PTAEHUFMBsGMHsC2lQBd5oBYoCoE8AHSAZVgCcBLA1UABWgEM8BzM+AVwDsATAGiwoBnUENANQAd0gAjQRVSQAUCEmYKsTKGYUAbpGF4OY0BoadYKdJMoL+gzAzIoz3UNEiPOofEVKVqAHSKymAAmkYI7NCuqGqcANag8ABmIjQUXrFOKBJMggBcISGgoAC0oACCbvCwDKgU8JkY7p7ehCTkVDQS2E6gnPCxGcwmZqDSTgzxxWWVoASMFmgYkAAeRJTInN3ymj4d-jSCeNsMq-wuoPaOltigAKoASgAywhK7SbGQZIIz5VWCFzSeCrZagNYbChbHaxUDcCjJZLfSDbExIAgUdxkUBIursJzCFJtXydajBBCcQQ0bh1cQAXlAAG9FCUSog8LQ2AR8mgyOwUAAyAWgABEQzemOg4xQBEo20g3BFvBZrPZAHkvmROfBuTiGNBBILhSL7BxuJwAOQ0aQyuUKRWKAC+5Magng7gC0HgzAAFAADdnagh0gAkjJpqAYAUDXMdfoAlABuRQUt0er2+gMcrmh8O06N4DW9INxpPBFQ4ZoeMheRDwPoMYHsGhfNxmZjsBjMSD8WDQdSJRvwPQiLwigCiqwYiAW+hFSWxIvADlQFuEADlIBIRUEVEXvu9Dfxu0dI2R7eM8KAnHWdMMsPoUAhENDhJdWxJ69ERAYONiFkwrAcDwQQlEAA